src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJava8Class1Ast.txt

```diff
@@ -292,412 +292,526 @@ CLASS_DEF -> CLASS_DEF [17:0]
-    |   |--TYPE -> TYPE [33:11]
-    |   |   `--LITERAL_VOID -> void [33:11]
-    |   |--IDENT -> instructions [33:16]
-    |   |--LPAREN -> ( [33:28]
-    |   |--PARAMETERS -> PARAMETERS [33:29]
-    |   |--RPAREN -> ) [33:29]
-    |   `--SLIST -> { [33:31]
-    |       |--VARIABLE_DEF -> VARIABLE_DEF [35:8]
-    |       |   |--MODIFIERS -> MODIFIERS [35:8]
-    |       |   |--TYPE -> TYPE [35:8]
-    |       |   |   `--LITERAL_BOOLEAN -> boolean [35:8]
-    |       |   |--IDENT -> b [35:16]
-    |       |   `--ASSIGN -> = [35:18]
-    |       |       `--EXPR -> EXPR [35:34]
-    |       |           `--GT -> > [35:34]
-    |       |               |--METHOD_CALL -> ( [35:31]
-    |       |               |   |--DOT -> . [35:24]
-    |       |               |   |   |--IDENT -> Math [35:20]
-    |       |               |   |   `--IDENT -> random [35:25]
-    |       |               |   |--ELIST -> ELIST [35:32]
-    |       |               |   `--RPAREN -> ) [35:32]
-    |       |               `--NUM_INT -> 0 [35:36]
-    |       |--SEMI -> ; [35:37]
+    |--METHOD_DEF -> METHOD_DEF [28:4]
+    |   |--MODIFIERS -> MODIFIERS [28:4]
+    |   |   `--LITERAL_PUBLIC -> public [28:4]
+    |   |--TYPE -> TYPE [28:11]
+    |   |   `--LITERAL_VOID -> void [28:11]
+    |   |--IDENT -> m6 [28:16]
+    |   |--LPAREN -> ( [28:18]
+    |   |--PARAMETERS -> PARAMETERS [28:19]
+    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [28:19]
+    |   |       |--MODIFIERS -> MODIFIERS [28:19]
+    |   |       |   `--ANNOTATION -> ANNOTATION [28:19]
+    |   |       |       |--AT -> @ [28:19]
+    |   |       |       `--IDENT -> Nullable [28:20]
+    |   |       |--TYPE -> TYPE [28:29]
+    |   |       |   |--IDENT -> String [28:29]
+    |   |       |   `--ANNOTATIONS -> ANNOTATIONS [28:36]
+    |   |       |       `--ANNOTATION -> ANNOTATION [28:36]
+    |   |       |           |--AT -> @ [28:36]
+    |   |       |           `--IDENT -> Nullable [28:37]
+    |   |       |--ELLIPSIS -> ... [28:46]
+    |   |       `--IDENT -> vararg [28:50]
+    |   |--RPAREN -> ) [28:56]
+    |   `--SLIST -> { [28:58]
+    |       `--RCURLY -> } [28:59]
+    |--METHOD_DEF -> METHOD_DEF [29:4]
+    |   |--MODIFIERS -> MODIFIERS [29:4]
+    |   |   `--LITERAL_PUBLIC -> public [29:4]
+    |   |--TYPE -> TYPE [29:46]
+    |   |   `--ARRAY_DECLARATOR -> [ [29:46]
+    |   |       |--ARRAY_DECLARATOR -> [ [29:33]
+    |   |       |   |--IDENT -> String [29:11]
+    |   |       |   |--ANNOTATIONS -> ANNOTATIONS [29:23]
+    |   |       |   |   `--ANNOTATION -> ANNOTATION [29:23]
+    |   |       |   |       |--AT -> @ [29:23]
+    |   |       |   |       `--IDENT -> Nullable [29:24]
+    |   |       |   `--RBRACK -> ] [29:34]
+    |   |       |--ANNOTATIONS -> ANNOTATIONS [29:36]
+    |   |       |   `--ANNOTATION -> ANNOTATION [29:36]
+    |   |       |       |--AT -> @ [29:36]
+    |   |       |       `--IDENT -> Nullable [29:37]
+    |   |       `--RBRACK -> ] [29:47]
```

Comment:
@pbludov  here is the new AST - does this look like what we want?

```
--TYPE -> TYPE [29:46]
   `--ARRAY_DECLARATOR -> [ [29:46]
       |--ARRAY_DECLARATOR -> [ [29:33]
       |   |--IDENT -> String [29:11]
       |   |--ANNOTATIONS -> ANNOTATIONS [29:23]
       |   |   `--ANNOTATION -> ANNOTATION [29:23]
       |   |       |--AT -> @ [29:23]
       |   |       `--IDENT -> Nullable [29:24]
       |   `--RBRACK -> ] [29:34]
       |--ANNOTATIONS -> ANNOTATIONS [29:36]
       |   `--ANNOTATION -> ANNOTATION [29:36]
       |       |--AT -> @ [29:36]
       |       `--IDENT -> Nullable [29:37]
       `--RBRACK -> ] [29:47]
```

Reply:
Yes, thank you. For `new String[][];` the AST is
```
`--LITERAL_NEW -> new [4:21]
    |--IDENT -> String [4:25]
    `--ARRAY_DECLARATOR -> [ [4:33]
        |--ARRAY_DECLARATOR -> [ [4:31]
        |   `--RBRACK -> ] [4:32]
        `--RBRACK -> ] [4:34]
```

For `new @First String @Second [] @Third @Fourth [];` the AST should be
```
`--LITERAL_NEW -> new [4:21]
    |--ANNOTATIONS -> ANNOTATIONS [4:25]
    |   `--ANNOTATION -> ANNOTATION [4:25]
    |       |--AT -> @ [4:25]
    |       `--IDENT -> First [4:26]
    |--IDENT -> String [4:32]
    `--ARRAY_DECLARATOR -> [ [4:65]
        |--ANNOTATIONS -> ANNOTATIONS [4:50]
        |   |--ANNOTATION -> ANNOTATION [4:50]
        |   |   |--AT -> @ [4:50]
        |   |   `--IDENT -> Third [4:51]
        |   `--ANNOTATION -> ANNOTATION [4:57]
        |       |--AT -> @ [4:57]
        |       `--IDENT -> Fourth [4:58]
        |--ARRAY_DECLARATOR -> [ [4:47]
        |   |--ANNOTATIONS -> ANNOTATIONS [4:39]
        |   |   `--ANNOTATION -> ANNOTATION [4:39]
        |   |       |--AT -> @ [4:39]
        |   |       `--IDENT -> Second [4:40]
        |   `--RBRACK -> ] [4:48]
        `--RBRACK -> ] [4:66]

```